modules/c++/sys/source/ExecWin32.cpp

```diff
@@ -23,10 +23,97 @@
 
 #if defined(WIN32)
 
-#include "sys/Err.h"
-#include "sys/Exec.h"
+#include <sys/Exec.h>
+#include <str/Manip.h>
 
-int sys::ExecPipe::closePipe()
+#define READ_PIPE  0
+#define WRITE_PIPE 1
+
+#include <fcntl.h>
+#include <io.h>
+typedef int pid_t;
+#define pipe(phandles)  _pipe(phandles, 4096, _O_BINARY)
+#define pclose _pclose
+#define fileno _fileno
+
+namespace sys
+{
+
+FILE* ExecPipe::openPipe(const std::string& command,
+                         const std::string& type)
+{
+    register FILE* ioFile;
```

Comment:
Why `register` keyword (both here and in Unix version)?  Don't think this is needed...